services/udp/service.go

```diff
@@ -146,27 +150,39 @@ func (s *Service) serve() {
 			return
 		default:
 			// Keep processing.
+			buf := make([]byte, UDPBufferSize)
+			n, _, err := s.conn.ReadFromUDP(buf)
+			if err != nil {
+				s.statMap.Add(statReadFail, 1)
+				s.Logger.Printf("Failed to read UDP message: %s", err)
+				continue
+			}
+			s.bytebuf <- buf[:n]
 		}
+	}
+}
 
-		n, _, err := s.conn.ReadFromUDP(buf)
-		if err != nil {
-			s.statMap.Add(statReadFail, 1)
-			s.Logger.Printf("Failed to read UDP message: %s", err)
-			continue
-		}
-		s.statMap.Add(statBytesReceived, int64(n))
+func (s *Service) parser() {
+	defer s.wg.Done()
 
-		points, err := models.ParsePoints(buf[:n])
-		if err != nil {
-			s.statMap.Add(statPointsParseFail, 1)
-			s.Logger.Printf("Failed to parse points: %s", err)
-			continue
-		}
+	for {
+		select {
+		case <-s.done:
+			return
+		case buf := <-s.bytebuf:
+			s.statMap.Add(statBytesReceived, int64(len(buf)))
```

Comment:
You're right about it being negligible, I moved it over here when I was fiddling with the performance and didn't see any change when I did it, but then I never put it back. I'll move it back because I agree that it's more readable that way.